### PR TITLE
support for yolo dataset loading and saving

### PR DESCRIPTION
**Pull request by SkalskiP**
# Description

YOLO Dataset loader is designed to be as flexible as possible to allow data consistency for both mixed and non-mixed datasets. You can test API in [this example notebook](https://colab.research.google.com/drive/1yyIFExkgks24JGX2jOf8TLAYTSIFIUhy?usp=sharing).

- box dataset

download - 2023-05-08T143022 501

- mask dataset

download - 2023-05-08T143026 472

- non-mixed datasets

download (100)

# API

- To load a consistent YOLO dataset. Only boxes or boxes and polygons. 

```python
dataset = sv.Dataset.from_yolo(
    images_directory_path=f"{dataset.location}/train/images",
    annotations_directory_path=f"{dataset.location}/train/labels",
    data_yaml_path=f"{dataset.location}/data.yaml",
    force_masks=False
)
```

- Force consistent mask loading from mixed-label YOLO dataset.

```python
dataset = sv.Dataset.from_yolo(
    images_directory_path=f"{dataset.location}/train/images",
    annotations_directory_path=f"{dataset.location}/train/labels",
    data_yaml_path=f"{dataset.location}/data.yaml",
    force_masks=True
)

### CODE REVIEW

**Review by Jacobsolawetz — APPROVED** [review]

File: supervision/dataset/core.py [hunk 1]
```diff
@@ -96,6 +97,28 @@ def from_pascal_voc(
...
+            >>> import supervision as sv
+
+            >>> roboflow.login()
+
+            >>> rf = Roboflow()
+
+            >>> project = rf.workspace(WORKSPACE_ID).project(PROJECT_ID)
+            >>> dataset = project.version(PROJECT_VERSION).download("voc")
+
+            >>> train_dataset = sv.Dataset.from_yolo(
```
Comment (Jacobsolawetz) [inline supervision/dataset/core.py:114]: 🔥 splits are separate sv Datasets?

Comment (SkalskiP) [reply supervision/dataset/core.py:114]: Yeah. You load every split as a separate `sv.Dataset` but that is pretty much the standard approach. Torch Datasets and Dataloaders work the same way. But I think we will build some utils to split and merge datasets to.
